pkg/machinery/resources/registry/registry.go

```diff
@@ -0,0 +1,144 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package registry provides resources for container mirror registry configuration.
+package registry
+
+import (
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+	"github.com/siderolabs/crypto/x509"
+
+	config2 "github.com/siderolabs/talos/pkg/machinery/config/config"
+	"github.com/siderolabs/talos/pkg/machinery/config/types/v1alpha1"
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+	"github.com/siderolabs/talos/pkg/machinery/resources/config"
+)
+
+//go:generate deep-copy -type RegistriesSpecSpec -header-file ../../../../hack/boilerplate.txt -o deep_copy.generated.go .
+
+// RegistriesSpecType is type of RegistriesSpec resource.
+const RegistriesSpecType = resource.Type("Registries.net.talos.dev")
```

Comment:
let's move this to `cri/` as it's the closest directory matching the purpose

Reply:
Done.